File: Suspension matlab/Suspension matlab raw/spring/ride_rate.m
%% ride rate
clear; clc; close all;

%% Parameters

MR = 0.9;                 % motion ratio
Kt = 180000;              % tire stiffness (N/m)

%% Spring stiffness range

Ks = linspace(10000,500000,2000);   % spring stiffness (N/m)

%% Wheel rate

Kw = Ks .* MR.^2;

%% Ride rate (series spring equation)

Kr = (Kw .* Kt) ./ (Kw + Kt);

%% Plot

figure;
plot(Ks/1000,Kr/1000,'LineWidth',2);

grid on;
xlabel('Spring Rate (kN/m)');
ylabel('Ride Rate (kN/m)');
title('Spring Rate vs Ride Rate');

%% Reference line: tire stiffness

hold on;
yline(Kt/1000,'--','Tire Stiffness');

legend('Ride Rate','Tire Stiffness');

File: Suspension matlab/spring/ride_rate/ride_rate.py
from pathlib import Path
import sys

import matplotlib.pyplot as plt
import numpy as np

ROOT = next(p for p in Path(__file__).resolve().parents if (p / "suspension_utils.py").exists())
sys.path.insert(0, str(ROOT))

from suspension_utils import load_data, show_plots


def main() -> None:
    p = load_data(Path(__file__).with_name("data.csv"))
    mr = p["ride_MR"]
    kt = p["ride_Kt"]
    ks = np.asarray(p["ride_Ks"], dtype=float)

    kw = ks * mr**2
    kr = (kw * kt) / (kw + kt)

    plt.figure()
    plt.plot(ks / 1000, kr / 1000, linewidth=2, label="Ride Rate")
    plt.axhline(kt / 1000, linestyle="--", label="Tire Stiffness")
    plt.grid(True)
    plt.xlabel("Spring Rate (kN/m)")
    plt.ylabel("Ride Rate (kN/m)")
    plt.title("Spring Rate vs Ride Rate")
    plt.legend()
    show_plots()


if __name__ == "__main__":
    main()
